src/assignment/views/AssignmentOverview.tsx

```diff
@@ -134,8 +127,24 @@ const AssignmentOverview: FunctionComponent<
 	const [markedAssignment, setMarkedAssignment] = useState<Avo.Assignment.Assignment | null>(
 		null
 	);
-	const [canEditAssignments, setCanEditAssignments] = useState<boolean | null>(null);
-	const [showPublicState, setShowPublicState] = useState<boolean | null>(null);
+	const canEditAssignments: boolean | null = useMemo(() => {
+		if (!commonUser) {
+			return null;
+		}
+		return (
+			PermissionService.hasPerm(commonUser, PermissionName.EDIT_ANY_ASSIGNMENTS) ||
+			PermissionService.hasPerm(commonUser, PermissionName.EDIT_OWN_ASSIGNMENTS)
+		);
```

Comment:
Not sure we ever want to handle null (loading?) states for permissions, if the user data isn't loaded that means they can't do permission-enabled actions imo.

```suggestion
		return commonUser && (
			PermissionService.hasPerm(commonUser, PermissionName.EDIT_ANY_ASSIGNMENTS) ||
			PermissionService.hasPerm(commonUser, PermissionName.EDIT_OWN_ASSIGNMENTS)
		);
```